uwblueprint/social-diversity-for-children
# feat(admin signup): invite teachers and admins internally

### Notion ticket link



[Sign In Flow - Admin Status](https://www.notion.so/uwblueprintexecs/Sign-In-Flow-Admin-Status-9d81ae5feb414b658652023aaabb9976)



### Implementation description

Combine Flow:

https://user-images.githubusercontent.com/44826218/141663807-b14605ea-d2c1-4863-a378-2e7657d7b704.mp4

```
Outdated Videos:
https://user-images.githubusercontent.com/44826218/141409020-ea7b6357-8525-45da-9f9f-29301cf5d0d1.mp4
https://user-images.githubusercontent.com/44826218/141409023-d6b13394-99c0-4893-ba6d-464b3bfdb9d5.mp4
```

- Add two new pages for each user invite for teacher and admin roles
  - Used Greg's text fields for forms on new pages
  - Create a user for the email with the given name and role, it's just that the user has not been verified yet
  - Use next-auth to signIn for the email, essentially sending magic links
  - Use toasts to indicate whether or not invite was a success
- Add new create user "POST" request for /api/user for internal user creation purposes
  - Create relevant utils to easily create users



### Steps to test

1. Log in as a program admin
2. Go to invite teacher page (route: /admin/add/teacher) - page should load without error
3. Enter an email that does not belong to any existing teacher - invite should be sent, the magic link sent should work or alternatively, logging in with that email automatically assigns you the teacher role such that you are able to access the admin platform
4. Enter an email that belongs to an existing to a user, there should be an error toast telling the admin that they cannot invite existing user
5. Step 4 applies to previously invited teacher
6. Repeat step 2-5 with program admins on the route /admin/add/program-admin
 


### What should reviewers focus on?

- Design: Look and feel of everything. I know this was done without any designs so I'm down to patch things to everyone's liking!
- Dev: Security and logic admin signup, especially with new user creation
- Dev: Code quality

### Checklist
